test/gateway/eip20_cogateway/progress_revert_redeem.js

```diff
@@ -0,0 +1,385 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+
+const EIP20CoGateway = artifacts.require("TestEIP20CoGateway");
+const MockUtilityToken = artifacts.require("MockUtilityToken");
+const BN = require("bn.js");
+
+const messageBus = require('../../test_lib/message_bus.js');
+const Utils = require("../../test_lib/utils.js");
+const web3 = require("../../test_lib/web3.js");
+const EventDecoder = require("../../test_lib/event_decoder");
+const coGatewayUtils = require('./helpers/co_gateway_utils.js');
+const StubData = require("../../../test/data/redeem_revoked_1.json");
+
+const ZeroBytes = Utils.ZERO_BYTES32;
+const MessageStatusEnum = messageBus.MessageStatusEnum;
+
+contract('EIP20CoGateway.progressRevertRedeem() ', function (accounts) {
+
+  let utilityToken, cogateway, revertRedeemParams, bountyAmount, penaltyAmount,
+    owner,
+    facilitator;
+
+  beforeEach(async function () {
+
+    owner = accounts[0];
+    facilitator = accounts[1];
+
+    revertRedeemParams = {
+      redeemer: StubData.co_gateway.redeem.params.redeemer,
+      amount: new BN(StubData.co_gateway.redeem.params.amount, 16),
+      beneficiary: StubData.co_gateway.redeem.params.beneficiary,
+      gasPrice: new BN(StubData.co_gateway.redeem.params.gasPrice, 16),
+      gasLimit: new BN(StubData.co_gateway.redeem.params.gasLimit, 16),
+      nonce: new BN(StubData.co_gateway.redeem.params.nonce, 16),
+      hashLock: StubData.co_gateway.redeem.params.hashLock,
+      rlpParentNodes: StubData.gateway.confirm_revert_redeem_intent.proof_data.storageProof[0].serializedProof,
+      blockHeight: new BN(StubData.gateway.confirm_revert_redeem_intent.proof_data.block_number, 16),
+      storageRoot: StubData.gateway.confirm_revert_redeem_intent.proof_data.storageHash,
+      messageStatus: MessageStatusEnum.Declared
```

Comment:
Is there no better way (an all tests) to access these values? Having fields more than 2 levels deep is not readable at all.

Reply:
Extracted common keys.